components/TaskCard/index.js

```diff
@@ -1,20 +1,62 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
-import { Tasks, Task, TaskListItem, Label, Labels } from "./TaskCard.styled";
+import {
+  Tasks,
+  Task,
+  TaskListItem,
+  Label,
+  Labels,
+  Checkbox,
+  Wrapper,
+  CrossedText,
+} from "./TaskCard.styled";
 
 export const TaskCard = ({ project, index }) => {
   const [visible, setVisible] = useState(false);
+  const [checked, setChecked] = useState(false);
+
+  const { projectTitle, tasks, projectDue, projectSize, id } = project;
+
+  useEffect(() => {
+    setChecked(project.status);
+  }, [project.status]);
+
   const projectClickHandler = () => {
     setVisible(!visible);
   };
-  console.log(project, index);
-  const { projectTitle, tasks, projectDue, projectSize } = project;
+
+  const onChangeCheckboxHandler = () => {
+    const newChecked = !checked;
+    setChecked(newChecked);
+
+    const updatedProjects = JSON.parse(
+      localStorage.getItem("projectsData")
+    ).map((p) => {
+      if (p.id === id) {
+        return { ...p, status: newChecked };
+      }
+      return p;
+    });
+
+    localStorage.setItem("projectsData", JSON.stringify(updatedProjects));
+  };
+
   return (
-    <TaskListItem onClick={projectClickHandler}>
-      <b>
-        {`P${index + 1}: `}
-        {projectTitle}
-      </b>
+    <TaskListItem>
+      <Wrapper>
+        <CrossedText onClick={projectClickHandler} checked={checked}>
+          {`P${index + 1}: `}
+          {projectTitle}
+        </CrossedText>
+        <label>
+          <Checkbox
+            type="checkbox"
+            id={`checkbox-${id}`}
+            checked={checked}
+            onChange={onChangeCheckboxHandler}
+          />
+        </label>
+      </Wrapper>
       {visible && (
         <Tasks>
           {tasks.map((task, index) => (

```

Comment:
Still using `index` for the `key` prop. ⚠️👿 
Read [the rules of keys](https://react.dev/learn/rendering-lists#rules-of-keys) and change your implementation accordingly.